gobblin-core/src/main/java/gobblin/instrumented/Instrumented.java

```diff
@@ -131,6 +138,85 @@ public static MetricContext getMetricContext(State state, Class<?> klazz, List<T
   }
 
   /**
+   * Copies a metric context and returns a new context with different name, and possibly different tags.
+   * @param context Context to copy.
+   * @param tags New tags to apply to context. Repeated keys will override old tags.
+   * @param name Name of the new context. If absent or empty, will modify old name by adding a random integer at the end.
+   * @return Copied context.
+   */
+  public static MetricContext copyMetricContext(MetricContext context, List<Tag<?>> tags, Optional<String> name) {
+
+    String newName = name.orNull();
+
+    if (Strings.isNullOrEmpty(newName)) {
+      int randomId = RAND.nextInt();
```

Comment:
Do you need to guarantee generating a unique Id every time? If so you may want to use UUID.